### PR TITLE
Convert more peer peer REST calls

### PR DESCRIPTION
**Pull request by klauspost**
## Description

* Clean up in general.
* Add JSON wrapper.
* Add slice wrapper.
* Add option to make handler return nil error if no connection is given, `IgnoreNilConn`.

Converts the following:

```
+	HandlerGetMetrics
+	HandlerGetResourceMetrics
+	HandlerGetMemInfo
+	HandlerGetProcInfo
+	HandlerGetOSInfo
+	HandlerGetPartitions
+	HandlerGetNetInfo
+	HandlerGetCPUs
+	HandlerServerInfo
+	HandlerGetSysConfig
+	HandlerGetSysServices
+	HandlerGetSysErrors
+	HandlerGetAllBucketStats
+	HandlerGetBucketStats
+	HandlerGetSRMetrics
+	HandlerGetPeerMetrics
+	HandlerGetMetacacheListing
+	HandlerUpdateMetacacheListing
+	HandlerGetPeerBucketMetrics
+	HandlerStorageInfo
+	HandlerConsoleLog
+	HandlerGetLocks
+	HandlerBackgroundHealStatus
+	HandlerGetLastDayTierStats
+	HandlerSignalService
+	HandlerGetBandwidth
```

@harshavardhana We should probably postpone this until after next release. Even though I've tested there could be subtle things.

## How to test this PR?

Most are related to health diag, so `mc support diag` and `mc admin info` apply, but other calls apply as well.

`mc admin logs` appear broken on the current version. Needs https://github.com/minio/madmin-go/pull/269 to work.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by klauspost** [comment]
Smells like there is a crash somewhere.

Edit: Yeah.. probably a type assertion. Fixed.

File: cmd/admin-handlers.go [hunk 1]
```diff
@@ -2004,49 +2003,70 @@ func (a adminAPIHandlers) ConsoleLogHandler(w http.ResponseWriter, r *http.Reque
 
 	// Avoid reusing tcp connection if read timeout is hit
 	// This is needed to make r.Context().Done() work as
-	// expected in case of read timeout
+	// expected in case of read timeout (????)
```
Comment (harshavardhana) [inline cmd/admin-handlers.go]: This logic is a bit dated, not sure if its still valid.

File: cmd/admin-handlers.go [hunk 2]
```diff
@@ -2004,49 +2003,70 @@ func (a adminAPIHandlers) ConsoleLogHandler(w http.ResponseWriter, r *http.Reque
...
 	w.Header().Set("Connection", "close")
 
 	setEventStreamHeaders(w)
```
Comment (harshavardhana) [inline cmd/admin-handlers.go:2009]: We only need to do this that's all no other HTTP headers needed.

File: cmd/peer-rest-common.go [hunk 3]
```diff
@@ -18,68 +18,29 @@
...
-	peerRESTMethodSysConfig                   = "/sysconfig"
-	peerRESTMethodDeleteBucketMetadata        = "/deletebucketmetadata"
-	peerRESTMethodLoadBucketMetadata          = "/loadbucketmetadata"
-	peerRESTMethodGetBucketStats              = "/getbucketstats"
-	peerRESTMethodGetAllBucketStats           = "/getallbucketstats"
 	peerRESTMethodVerifyBinary                = "/verifybinary"
 	peerRESTMethodCommitBinary                = "/commitbinary"
 	peerRESTMethodSignalService               = "/signalservice"
 	peerRESTMethodBackgroundHealStatus        = "/backgroundhealstatus"
 	peerRESTMethodGetLocks                    = "/getlocks"
```
Comment (harshavardhana) [inline cmd/peer-rest-common.go]: These can be converted as well

File: cmd/peer-rest-common.go [hunk 4]
```diff
@@ -18,68 +18,29 @@
...
-	peerRESTMethodGetLocalDiskIDs             = "/getlocaldiskids"
 	peerRESTMethodGetBandwidth                = "/bandwidth"
-	peerRESTMethodGetMetacacheListing         = "/getmetacache"
-	peerRESTMethodUpdateMetacacheListing      = "/updatemetacache"
-	peerRESTMethodGetPeerMetrics              = "/peermetrics"
-	peerRESTMethodGetPeerBucketMetrics        = "/peerbucketmetrics"
-	peerRESTMethodLoadTransitionTierConfig    = "/loadtransitiontierconfig"
 	peerRESTMethodSpeedTest                   = "/speedtest"
 	peerRESTMethodDriveSpeedTest              = "/drivespeedtest"
 	peerRESTMethodReloadSiteReplicationConfig = "/reloadsitereplicationconfig"
```
Comment (harshavardhana) [inline cmd/peer-rest-common.go]: and these as well, they do not send any data.

File: internal/grid/handlers.go [hunk 5]
```diff
@@ -409,6 +452,17 @@ func (h *SingleHandler[Req, Resp]) AllowCallRequestPool(b bool) *SingleHandler[R
...
 }
 
+// IgnoreNilConn will ignore nil connections when calling.
+// This will make Call return nil instead of ErrDisconnected when the connection is nil.
+// This may only be set ONCE before use.
+func (h *SingleHandler[Req, Resp]) IgnoreNilConn() *SingleHandler[Req, Resp] {
+	if h.ignoreNilConn {
+		logger.LogOnceIf(context.Background(), fmt.Errorf("%s: IgnoreNilConn called twice", h.id.String()), h.id.String()+"IgnoreNilConn")
+	}
+	h.ignoreNilConn = true
```
Comment (harshavardhana) [inline internal/grid/handlers.go:466]: what is the use-case for this?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/admin-handlers.go (hunk 1 above)
Comment (klauspost) [reply cmd/admin-handlers.go]: Yeah. Seems outdated.

File: internal/grid/handlers.go (hunk 5 above)
Comment (klauspost) [reply internal/grid/handlers.go:466]: This allows removing the `if conn == nil {return nil}` - hit when grid has not yet been initialized - in the client callers. See the changes in all the existing rpc calls.

File: cmd/peer-rest-common.go (hunk 4 above)
Comment (klauspost) [reply cmd/peer-rest-common.go]: I was hesitant for the speedtests.. but I guess the information only ones would be fine.

Comment (klauspost) [reply cmd/peer-rest-common.go]: Took the Reload handler.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**